pkg/kudoctl/cmd/uninstall.go

```diff
@@ -0,0 +1,130 @@
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/kudobuilder/kudo/pkg/kudoctl/clog"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/env"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/util/kudo"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	uninstallExample = `kubectl kudo uninstall --instance flink`
+)
+
+type uninstallOptions struct {
+	InstanceName string
+	Purge        bool
+}
+
+type uninstallCmd struct{}
+
+func (cmd *uninstallCmd) run(options uninstallOptions, settings *env.Settings) error {
+	kc, err := kudo.NewClient(settings.Namespace, settings.KubeConfig)
+	clog.V(3).Printf("acquiring kudo client")
+	if err != nil {
+		clog.V(3).Printf("failed to acquire kudo client")
+		return fmt.Errorf("failed to acquire kudo client: %w", err)
+	}
+
+	return cmd.uninstall(kc, options.InstanceName, options.Purge, settings)
+}
+
+func (cmd *uninstallCmd) uninstall(kc *kudo.Client, instanceName string, purge bool, settings *env.Settings) error {
+	instance, err := kc.GetInstance(instanceName, settings.Namespace)
+	if err != nil {
+		return fmt.Errorf("failed to verify if instance already exists: %w", err)
+	}
+
+	if instance == nil {
+		return fmt.Errorf("instance %s in namespace %s does not exist in the cluster", instanceName, settings.Namespace)
+	}
+
+	operatorVersionName := instance.Spec.OperatorVersion.Name
+	operatorVersionNamespace := settings.Namespace
+	if instance.Spec.OperatorVersion.Namespace != "" {
+		operatorVersionNamespace = instance.Spec.OperatorVersion.Namespace
+	}
+
+	err = kc.DeleteInstance(instanceName, settings.Namespace)
+	if err != nil {
+		return err
+	}
+
+	if purge {
+		err = cmd.deleteOperatorVersionAndOperator(kc, operatorVersionName, operatorVersionNamespace)
+		if err != nil {
+			return fmt.Errorf("failed to delete operatorversion and operator of instance %s: %w", instanceName, err)
+		}
+	}
+
+	return nil
+}
+
+func (cmd *uninstallCmd) deleteOperatorVersionAndOperator(kc *kudo.Client, operatorVersionName, namespace string) error {
+	operatorVersion, err := kc.GetOperatorVersion(operatorVersionName, namespace)
```

Comment:
probably some more verbose logging would be nice so that it would be seen from cmdline "we have now deleted OV xxx" etc.

Reply:
Yes, agree, we should have feedback similar to `install` about what resources have been deleted.